src/basics/clue-result.js

```diff
@@ -39,17 +39,48 @@ export function elim_result(player, hypo_player, hand, list) {
 
 /**
  * @param  {Game} game
+ * @param  {Game} hypo_game
  * @param  {Player} hypo_player
+ * @param  {number} giver
  * @param  {number} target
  * @param  {number} focus_order
  */
-export function bad_touch_result(game, hypo_player, target, focus_order = -1) {
-	const { me, state } = game;
-	let bad_touch = 0, trash = 0;
+export function bad_touch_result(game, hypo_game, hypo_player, giver, target, focus_order = -1) {
+	const { me, state } = hypo_game;
+	let bad_touch = 0, trash = 0, avoidable_dupe = 0;
+
+	const dupe_scores = game.players.map((player, pi) => {
+		if (pi == target)
+			return Infinity;
+		let possible_dupe = 0;
+		// Check if the giver may have a touched duplicate card.
+		// TODO: Should we consider chop moved cards?
+		for (const card of state.hands[target]) {
+			const identity = card.identity();
+			// TODO: Should we cluing cards where receiver knows they are duplicated?
+			if (!identity || !hypo_game.state.isPlayable(identity))
```

Comment:
Why does it matter if the card is playable?

Also, if we're using `card.identity()` from the state's hands, they'll all be `ActualCard`s from the deck and thus will be visible, as long as they aren't in our hand. So the identity may be playable even if it's not common knowledge.